### ISSUE TITLE
BigtableSession closes provided ExecutorService on #close() which is prone to deadlocks

### ISSUE THREAD

**Issue by udoprog** [GoogleCloudPlatform/cloud-bigtable-client#393]
I'm attempting to incorporate BigtableSession into a service which uses a shared ExecutorService for common tasks and establishes multiple bigtable backends.

Currently I am required to create an ExecutorService per BigtableSession instance because the `BigtableSession#close()` method aggressively attempts to assert that the provided ExecutorService has been shut down.

I initially ended up queuing up a task to close the `BigtableSession` on the same thread pool, causing `BigtableSession#close()` to either deadlock (or wait for a long time) on a [`ExecutorService#isTerminated()`](https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/master/bigtable-grpc-interface/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java#L474) loop (see test below).

I believe the either **1)** behavior should be modified so that `BigtableSession` does not shutdown a user-provided `ExecutorService`, or **2)** it does not allow the user to provide a thread pool at all.

## Test Case

This test case always deadlocks for me.

```java
public class BigtableSessionTest {
    public static final String USER_AGENT = "test";

    public static void main(String[] args) throws Exception {
        CredentialOptions credentials = /* ... */;
        final MySession builder = /* ... */;

        final ExecutorService executorService = Executors.newFixedThreadPool(4);

        final Future<Void> future = executorService.submit(new Callable<Void>() {
            @Override
            public Void call() throws Exception {
                // attempt to close on the same thread pool...
                builder.build(executorService).close();
                return null;
            }
        });

        try {
            future.get(30, TimeUnit.SECONDS);
        } catch (TimeoutException e) {
            System.out.println("Timeout :(");
            e.printStackTrace();
        }

        final ExecutorService executorService2 = Executors.newFixedThreadPool(4);

        final BigtableSession session = builder.build(executorService2);

        System.out.println("Main closing...");
        session.close();
        System.out.println("Closed without problems!");

        System.out.println("bye, bye");
        System.exit(0);
    }

    static class MySession {
        final String project;
        final String zone;
        final String cluster;
        final CredentialOptions credentials;

        public MySession(final String project, final String zone, final String cluster, CredentialOptions credentials) {
            this.project = project;
            this.zone = zone;
            this.cluster = cluster;
            this.credentials = credentials;
        }

        public BigtableSession build(final ExecutorService executorService) throws IOException {
            final BigtableOptions options = new BigtableOptions.Builder().setProjectId(project).setZoneId(zone)
                    .setClusterId(cluster).setUserAgent(USER_AGENT).setCredentialOptions(credentials).build();

            return new BigtableSession(options, executorService);
        }
    }
}
```

## Output

```
Aug 07, 2015 6:36:21 PM com.google.cloud.bigtable.config.Logger info
INFO: Opening connection for ...
Timeout :(
java.util.concurrent.TimeoutException
	at java.util.concurrent.FutureTask.get(FutureTask.java:205)
	at com.spotify.bigtable.BigtableSessionTest.main(BigtableSessionTest.java:40)
Aug 07, 2015 6:36:51 PM com.google.cloud.bigtable.config.Logger info
INFO: Opening connection for ...
Main thread closeing...
Closed without problems!
bye, bye
```

**Comment by sduskis** [comment]
I'll fix this in master before releasing the 0.2.0 version of the client.  FYI, we have a package refactoring in progress that will affect you: https://github.com/GoogleCloudPlatform/cloud-bigtable-client/pull/391

**Comment by udoprog** [comment]
Thanks for the heads up!
